src/enums/AuthenticationStates.js

```diff
@@ -0,0 +1,9 @@
+import {Enum} from 'enumify';
+
+export default class AuthenticationStates extends Enum {}
+AuthenticationStates.initEnum([
+  'AUTHENTICATED',
+  'CONFIRMED',
+  'REJECTED',
```

Comment:
I would propose instead:

* `AUTHENTICATED` (meaning fully authenticated and confirmed)
* `PENDING_CONFIRMATION`
* `UNAUTHENTICATED`

Allows us to undo a bunch of the `authenticated` -> `confirmed` changes in the PR (I think `isUserAuthenticated` is generally more descriptive than `isUserConfirmed`) and keeps the concern of confirmation more local to just the confirmation process